tidy3d/plugins/design/method.py

```diff
@@ -175,28 +110,466 @@ class MethodGrid(MethodIndependent):
+
+        # Store parameters and fitness
+        store_parameters = []
+        store_fitness = []
+        store_aux = []
+
+        # Set gene_spaces to keep GA within ranges
+        param_converter = {}
+        gene_spaces = []
+        gene_types = []
+        for param in parameters:
+            if type(param) == ParameterFloat:
+                gene_spaces.append({"low": param.span[0], "high": param.span[1]})
+                gene_types.append(float)
+            elif type(param) == ParameterInt:
+                gene_spaces.append(
+                    range(param.span[0], param.span[1] + 1)
+                )  # +1 included so as to be inclusive of upper range value
+                gene_types.append(int)
+            else:
+                # Designed for str in ParameterAny but may work for anything
+                param_converter[param.name] = self.any_to_int_param(param)
+
+                gene_spaces.append(range(0, len(param.allowed_values)))
+                gene_types.append(int)
+
+        # Create fitness function combining pre and post fn with the tidy3d call
+        def fitness_function(ga_instance, solution, solution_idx):
+            # Break solution down to dict
+            sol_dict = self.sol_array_to_dict(solution, param_keys, param_converter)
+
+            # Iterate through solutions for batched and non-batched data
+            sol_out, aux_out = self._extract_output(run_fn(sol_dict))
+
+            self._flatten_and_append(aux_out, store_aux)
+
+            return sol_out
+
+        def on_generation(ga_instance):
+            store_parameters.append(ga_instance.population.copy())
+            store_fitness.append(ga_instance.last_generation_fitness)
+            if console is not None:
+                best_fitness = ga_instance.best_solution()[1]
+                console.log(
+                    f"Generation {ga_instance.generations_completed} Best Fitness: {best_fitness:.3f}"
+                )
+
+        # Determine initial array
+        num_genes = len(parameters)
+
+        # define the optimizer
+        ga_instance = pygad.GA(
+            num_generations=self.n_generations,
+            num_parents_mating=self.n_parents_mating,
+            fitness_func=fitness_function,
+            parent_selection_type=self.parent_selection_type,
+            mutation_type=self.mutation_type,
+            mutation_probability=self.mutation_prob,
+            crossover_type=self.crossover_type,
+            crossover_probability=self.crossover_prob,
+            sol_per_pop=self.solutions_per_pop,
+            num_genes=num_genes,
+            fitness_batch_size=self.solutions_per_pop,
+            on_generation=on_generation,
+            random_seed=self.rng_seed,
+            gene_space=gene_spaces,
+            gene_type=gene_types,
+            stop_criteria=self.stop_criteria,
+            save_solutions=True,
+        )
+
+        ga_instance.run()
+
+        if console is not None:
```

Comment:
might want this to also happen only if verbose is True in the design.. if it's possible.

Reply:
The way it's setup currently console can only not be None when Verbose is True so I figured it wasn't worth passing verbose through to method when it can just check the state of console.

Reply:
oh interesting. that works